buildscripts/build_sodium_windows.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+#    Copyright © 2021 by The qTox Project Contributors
+#
+#    This program is libre software: you can redistribute it and/or modify
+#    it under the terms of the GNU General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+set -euo pipefail
+
+usage()
+{
+    echo "Download and build sodium for the windows cross compiling environment"
+    echo "Usage: $0 --arch {x86_64|i686}"
+}
+
+ARCH=""
+
+while (( $# > 0 )); do
+    case $1 in
+        --arch) ARCH=$2; shift 2 ;;
+        -h|--help) usage; exit 1 ;;
+        *) echo "Unexpected argument $1"; usage; exit 1;;
+    esac
+done
+
+if [ "$ARCH" != "i686" ] && [ "$ARCH" != "x86_64" ]; then
+    echo "Unexpected arch $ARCH"
+    usage
+    exit 1
+fi
```

Comment:
All the argument checking and whatnot in new build scripts is a lot of copy-pasta, would be nice to put it away in one place.

Reply:
Yeah I agree, unfortunately I don't have a good answer for how to fix this. Any API I could think of for exporting a list of environment variables was either more complex or less clear than just doing another case statement. I think the argument parsing kinda doubles as documentation for what's expected from a script so having it plainly laid out in each script has benefit.

I'm open to suggestions but this is the best I've come up with.

Reply:
Just move the arch check in a `windows_common.sh` snippet and just do `. windows_common.sh` in the scripts.

If you don't like `ARCH` variable magically appearing like that, you could move them into a `get_arch_arg` function inside `windows_common.sh` and do

```bash
. windows_common.sh
ARCH="$(get_arch_arg)"
```

or something like that, so that it's explicitly defined (but then `get_arch_arg` is magically appearing out of nowhere lol).

Reply:
Just as a note, as part of building macOS deps from source to allow setting `mmacosx-version-min` so that our app can be run on older systems than our build system, I worked on top of this and did basically that. It's still WIP, but currently it's using all the windows build scripts (which will need to be renamed) and setting environment variables based on macos/winx86_64/wini686.

Overall the PR needs cleanup, but the general idea is here: https://github.com/qTox/qTox/pull/6438/files#diff-21995f4949ecff10f7b50a4d2023a8cf7be90d670b8ca1fde8df1c8595c38b63

I kind of thought of it like all the build_x_windows.sh scripts are implementing an interface, so their API is common and doesn't need to be documented in each one, and those shared environment variables are part of that architecture. Obviously there's no actually in code anywhere because bash.. but it feels like the buildscripts are simple enough that it shouldn't be too confusing to figure out where that's coming from.

I agree it's still weird to have all this variables popping in, but I just kind of view that as the way bash is :S